### ISSUE TITLE
Overriding Default Serializer for typeof(object) programmatically

### ISSUE THREAD

**Issue by mchandschuh** [akkadotnet/akka.net#4761]
I'm trying to override the default `NewtonsoftJsonSerializer` with my own implementation of `Akka.Serialization.Serializer` and I want this to be done programmatically as it's in a base library and I really don't want all of my apps to need this extra bit of a config. I'm initializing my `ActorSystem` by creating a `SerializationSetup` that creates my custom serializer and instructs it via `SerializerDetails` to register under `typeof(object)` -- expected behavior is that this would *overwrite* the default `NewtonsoftJsonSerializer` but alas, it doesn't.

If running the attached sample in debug mode, you can break after the `ActorSystem` is created and inspect `system.Serialization._serializerMap` and observe the `typeof(object)` mapping still pointing to `NewtonsoftJsonSerializer`. There is a mapping for my custom `World` type but my serializer isn't even being invoked for that type. If I manually invoke `system.Serializer.AddSerializerMap(typeof(object), new CustomSerializer(...))` then it does start working.

- Which Akka.Net version you are using
> 1.4.16

- On which platform you are using Akka.Net
> dotnetcore v3.1 running on win10

- A list of steps to reproduce the issue. Or an gist or github repo which can be easily used to reproduce your case.
Please see this repository for a working example and comments/discussion on the observed behavior:
> https://github.com/mchandschuh/default-serializer-programmatic-override

**Comment by Aaronontheweb** [comment]
@mchandschuh added this to our Feb 8th bug triage sprint.

**Comment by Arkatufus** [comment]
I tried your example code on the latest Akka source and they worked as intended.

**Comment by mchandschuh** [comment]
@Arkatufus -- has the version this was tested against been released? I'm running the [example repository](https://github.com/mchandschuh/default-serializer-programmatic-override) code against 1.4.16 and the custom serializer isn't being called. Only after uncommenting this line does the custom serializer get invoked:
```
system.Serialization.AddSerializationMap(typeof(object), new CustomSerializer(system.Serialization.System));
```

If it hasn't been released yet, what will the version number be when it does get released?

Thank you!

**Comment by Arkatufus** [comment]
You're right, there was a bug in the serialization setup order making any user defined serializer that is already defined in the HOCON config to be overriden by the HOCON config settings. Let me make a quick fix for this.

**Comment by Arkatufus** [comment]
Fix is proposed in PR #4807
